app/models/user.rb

```diff
@@ -143,6 +143,17 @@ def can_use_inter_ressort?
     work_at_bex? && organization.use_inter_ressort?
   end
 
+  def too_many_appointments_without_status?
+    recent_past_booked_appointments_count = appointments
```

Comment:
Proposition (on a accès à convict.appointments):
Aussi, ça pourrait 
```
def recent_past_booked_appointments
   convict.appointments.joins(:slot).where(state: 'booked').where('slots.date >= ? AND slots.date < ?', 3.months.ago, Date.today)
end

def too_many_appointments_without_status?
  recent_past_booked_appointments.count > 5
end
```

Reply:
Je suis pas certain de comprendre. On est dans user.rb là, donc j'ai pas self.convict.appointments. Un user peut avoir plusieurs convicts.

Reply:
Ah yes my bad ! J'ai lu trop vite, tu as raison !

Reply:
Mais du coup pour pas te faire chier, ma proposition devient : rajouter un `has_many :appointments, through: : convicts` dans user.rb. Ca me semble utile est plus simple

Reply:
Ouais mais ça marche pas trop dans notre cas car le convict est pas vraiment une jointure standard avec un double belongs_to comme indiqué ici https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association. D'ailleurs ça fait péter des tests, notamment sur la suppression d'organization et de user

Reply:
C'est étonnant que ça ne marche pas. On est dans un cas ultra classique bien décrit par la doc d'ailleurs: "The has_many :through association is also useful for setting up "shortcuts" through nested has_many associations. For example, if a document has many sections, and a section has many paragraphs, you may sometimes want to get a simple collection of all paragraphs in the document.". Si tu veux on peut se faire un call rapide pour regarder.